src/containers/EditProfileContainer.js

```diff
@@ -16,20 +20,46 @@ const EditProfileContainer = props => {
     myUserInfo,
     uploadProfileImage,
   } = props
+
+  const [username, setUsername] = useState('')
+
+  useEffect(() => {
+    async function fetchUser() {
+      const user = await props.getUser(mmuserId)
+      setUsername(user.data.username)
+    }
+    fetchUser()
+  }, [])
+
   // Update profile picture
   const updatePicture = img => {
+    if (!img) {
```

Comment:
This is a little problematic, since in the edit form, there is never any picture on default. So if user edits anything and does not choose a new profile pic, then the default profile pic will be set.

Reply:
I wonder if this could be fixed by adding a boolean attribute to node user (profilepicUploaded or something) which could be used here (if nickname && !profilepicUploaded would change the picture and the username would also be updated, and nickname && profilePicUploaded would only update the nickname)

Reply:
So I've done this. Note that this needs backend/updateUsername branch to work.